components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/CacheBackedOrganizationManagementDAO.java

```diff
@@ -445,6 +447,43 @@ public List<AncestorOrganizationDO> getAncestorOrganizations(String organization
         return organizationMgtDAO.getAncestorOrganizations(organizationId);
     }
 
+    @Override
+    public BasicOrganization getBasicOrganization(String organizationId, String associatedTenantDomain)
+            throws OrganizationManagementException {
+
+        String tenantDomain = associatedTenantDomain;
+        if (tenantDomain == null) {
+            tenantDomain = resolveTenantDomain(organizationId);
+        }
+
+        OrganizationIdCacheKey cacheKey = new OrganizationIdCacheKey(organizationId);
+        BasicOrganizationCacheEntry entry = BasicOrganizationCacheByOrgId.getInstance()
+                .getValueFromCache(cacheKey, tenantDomain);
+
+        if (entry != null) {
+            LOG.debug("Basic Organization Cache entry found for organization id: " + organizationId);
+            return entry.getBasicOrganization();
+        }
+        LOG.debug("Basic Organization Cache entry not found for organization id: " + organizationId +
+                ". Fetching entry from DB.");
+
+        Map<String, BasicOrganization> orgMap = organizationMgtDAO.getBasicOrganizationDetailsByOrgIDs(
```

Comment:
doesn't this dao has a method to retrieve by id ?

Reply:
https://github.com/wso2/identity-organization-management-core/pull/194/commits/89824b49512ce1bbde2edc2412f0f790faa0df69